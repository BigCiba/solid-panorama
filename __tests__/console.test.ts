import { describe, expect, test } from '@jest/globals';
import {
    format,
    formatx
} from '../packages/panorama-polyfill/src/utils/format';

describe('console', function () {
    test('format', function () {
        const style = {
            paneltype: '',
            rememberchildfocus: '',
            SetPanelEvent: '',
            RunScriptInPanelContext: ''
        };
        const s = Symbol('s');
        const int = BigInt(99999);

        expect(
            format('test: %s %d %j', 'str', 123, false, 'test', { a: 123 })
        ).toMatchSnapshot();

        expect(
            formatx('test: %s %d %j', 'str', 123, false, 'test', { a: 123 })
        ).toMatchSnapshot();

        expect(
            format({
                x: 1,
                y: '123',
                c: 'str',
                d: false,
                e: undefined,
                f: null,
                g: function () {},
                h: () => {},
                i: test,
                style
            })
        ).toMatchSnapshot();

        expect(
            format({
                obj: { x: 1, y: 2, c: [123, { b: null }], style },
                ary: [1, 2, 3, 'A', false, null, [4, 56, { g: 1, c: [] }]]
            })
        ).toMatchSnapshot();

        expect(
            formatx(
                {
                    x: 1,
                    y: '123',
                    c: 'str',
                    d: false,
                    e: undefined,
                    f: null,
                    g: function () {},
                    h: () => {},
                    i: test,
                    style,
                    s,
                    int
                },
                true
            )
        ).toMatchSnapshot();

        expect(
            formatx(
                {
                    obj: {
                        x: 1,
                        y: 2,
                        style: {},
                        c: [123, { b: null, a: [null] }, [null]]
                    },
                    ary: [
                        1,
                        2,
                        3,
                        'A',
                        false,
                        null,
                        [4, 56, { g: 1, c: [{ c: 45 }] }]
                    ]
                },
                true
            )
        ).toMatchSnapshot();
    });
});
