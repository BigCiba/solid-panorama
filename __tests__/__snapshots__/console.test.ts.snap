// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`console format 1`] = `"test: str 123 false test { a: 123 }"`;

exports[`console format 2`] = `"test: 5 5 5 str 123 false test { a: 123 }"`;

exports[`console format 3`] = `"{ x: 1, y: "123", c: "str", d: false, e: undefined, f: null, g: [Function], h: [Function], i: [Function], style: [VCSSStyleDeclaration] }"`;

exports[`console format 4`] = `"{ obj: { x: 1, y: 2, c: [ 123, { b: null } ], style: [VCSSStyleDeclaration] }, ary: [ 1, 2, 3, "A", false, null, [ 4, 56, { g: 1, c: [  ] } ] ] }"`;

exports[`console format 5`] = `
"{
    x: 1,
    y: "123",
    c: "str",
    d: false,
    e: undefined,
    f: null,
    g: [Function],
    h: [Function],
    i: [Function],
    style: [VCSSStyleDeclaration],
    s: Symbol(s),
    int: [bigint 99999]
} true"
`;

exports[`console format 6`] = `
"{
    obj: {
        x: 1,
        y: 2,
        style: {

        },
        c: [
            123,
            {
                b: null,
                a: [
                    null
                ]
            },
            [
                null
            ]
        ]
    },
    ary: [
        1,
        2,
        3,
        "A",
        false,
        null,
        [
            4,
            56,
            {
                g: 1,
                c: [
                    {
                        c: 45
                    }
                ]
            }
        ]
    ]
} true"
`;
